pkg/kubectl/cmd/delete/delete.go

```diff
@@ -216,6 +219,34 @@ func (o *DeleteOptions) Validate() error {
 	case o.ForceDeletion:
 		fmt.Fprintf(o.ErrOut, "warning: --force is ignored because --grace-period is not 0.\n")
 	}
+
+	if o.WarnClusterScope {
+		infos,_ := o.Result.Infos()
+		clsRes := make(map[string]bool)
+		if infos != nil {
+			for _, info := range infos {
+				if info.Mapping.Scope.Name() == meta.RESTScopeNameRoot {
+					switch info.Name {
```

Comment:
don't special-case namespace names

Reply:
Done.
Case has been added  in test/cmd/core.sh, test/cmd/rbac.sh and test/cmd/storage.sh.